### PR TITLE
Send EnvVars to TestHost process while bootstrapping process.

### PR DESCRIPTION
**Pull request by harshjain2**
Send EnvVars to TestHost process while bootstrapping process.

Related to : https://github.com/microsoft/vstest/issues/309

### CODE REVIEW

File: src/Microsoft.TestPlatform.CrossPlatEngine/Helpers/Interfaces/IProcessHelper.cs [hunk 1]
```diff
@@ -18,8 +19,9 @@ internal interface IProcessHelper
         /// <param name="arguments">The command-line arguments.</param>
         /// <param name="workingDirectory">The working directory for this process.</param>
         /// <param name="exitCallback">Call back for on process exit</param>
+        /// <param name="environmentVariables">Environment variables to set while bootstrapping the process.</param>
         /// <returns>The process created.</returns>
-        Process LaunchProcess(string processPath, string arguments, string workingDirectory, Action<Process, string> errorCallback);
+        Process LaunchProcess(string processPath, string arguments, string workingDirectory, Action<Process, string> errorCallback, IDictionary<string,string> environmentVariables);
```
Comment (singhsarab) [inline src/Microsoft.TestPlatform.CrossPlatEngine/Helpers/Interfaces/IProcessHelper.cs]: Arrange the order.

File: src/Microsoft.TestPlatform.CrossPlatEngine/Helpers/ProcessHelper.cs [hunk 2]
```diff
@@ -39,6 +40,15 @@ public Process LaunchProcess(string processPath, string arguments, string workin
                 process.StartInfo.RedirectStandardError = true;
                 process.EnableRaisingEvents = true;
 
+                if (envVariables != null)
+                {
+                    var readonlyEnvVar = process.StartInfo.Environment;
```
Comment (singhsarab) [inline src/Microsoft.TestPlatform.CrossPlatEngine/Helpers/ProcessHelper.cs]: Nit: refactor

**Review by singhsarab — APPROVED** [review]
👍

File: src/Microsoft.TestPlatform.CrossPlatEngine/Helpers/Interfaces/IProcessHelper.cs (hunk 1 above)
Comment (harshjain2) [inline src/Microsoft.TestPlatform.CrossPlatEngine/Helpers/Interfaces/IProcessHelper.cs]: done.

File: src/Microsoft.TestPlatform.CrossPlatEngine/Helpers/ProcessHelper.cs (hunk 2 above)
Comment (harshjain2) [inline src/Microsoft.TestPlatform.CrossPlatEngine/Helpers/ProcessHelper.cs]: done.

**Pull request merged**